src/syscheckd/whodata/syscheck_audit.c

```diff
@@ -179,23 +171,93 @@ int init_auditd_socket(void) {
     return sfd;
 }
 
-void audit_no_rules_to_realtime() {
-    int found;
+void audit_create_rules_file() {
     char *real_path = NULL;
     int i;
+    FILE *fp;
+
+    fp = fopen(AUDIT_RULES_FILE, "w");
+    if (!fp) {
+        merror(FOPEN_ERROR, AUDIT_RULES_FILE, errno, strerror(errno));
+        return;
+    }
 
     for (i = 0; syscheck.dir[i] != NULL; i++) {
         if ((syscheck.opts[i] & WHODATA_ACTIVE) == 0) {
             continue;
         }
         real_path = fim_get_real_path(i);
-        found = search_audit_rule(real_path, WHODATA_PERMS, AUDIT_KEY);
 
-        if (found == 0) {   // No rule found
+        mdebug2(FIM_ADDED_RULE_TO_FILE, real_path);
+        fprintf(fp, "-w %s -p wa -k %s\n", real_path, AUDIT_KEY);
+
+        free(real_path);
+    }
+
+    if (fclose(fp)) {
+        merror(FCLOSE_ERROR, AUDIT_RULES_FILE, errno, strerror(errno));
+        return;
+    }
+
+    // Create symlink to audit rules file
+    if (symlink(AUDIT_RULES_FILE, AUDIT_RULES_LINK) < 0) {
+        switch (errno) {
+        case EEXIST:
+            if (unlink(AUDIT_RULES_LINK) < 0) {
+                merror(UNLINK_ERROR, AUDIT_RULES_LINK, errno, strerror(errno));
+                return;
+            }
+
+            if (symlink(AUDIT_RULES_FILE, AUDIT_RULES_LINK) == 0) {
+                break;
+            }
+
+        // Fallthrough
+        default:
+            merror(LINK_ERROR, AUDIT_RULES_LINK, AUDIT_RULES_FILE, errno, strerror(errno));
+            return;
+        }
+    }
+
+    mdebug2(FIM_AUDIT_CREATED_RULE_FILE);
+}
+
+void audit_rules_to_realtime() {
+    char *real_path = NULL;
+    int found;
+
+    for (int i = 0; syscheck.dir[i] != NULL; i++) {
+        if ((syscheck.opts[i] & WHODATA_ACTIVE) == 0) {
+            continue;
+        }
+
+        found = 0;
+        real_path = fim_get_real_path(i);
+
+        // Initialize audit_rule_list
+        int auditd_fd = audit_open();
+        int res = audit_get_rule_list(auditd_fd);
+        audit_close(auditd_fd);
+        if (!res) {
+            merror(FIM_ERROR_WHODATA_READ_RULE); // LCOV_EXCL_LINE
+        }
+
+        if (search_audit_rule(real_path, WHODATA_PERMS, AUDIT_KEY) == 1) {
+            found = 1;
+        } else {
+            for (int j = 0; syscheck.audit_key[j]; j++) {
+                if (search_audit_rule(real_path, WHODATA_PERMS, syscheck.audit_key[j]) == 1) {
+                    found = 1;
+                }
+            }
+        }
```

Comment:
I think unfolding these ifs could be easier to understand and maintain:
```suggestion
        if (search_audit_rule(real_path, WHODATA_PERMS, AUDIT_KEY) == 1) {
            free(real_path);
            continue;
        }

        for (int j = 0; syscheck.audit_key[j]; j++) {
            if (search_audit_rule(real_path, WHODATA_PERMS, syscheck.audit_key[j]) == 1) {
                found = 1;
                break;
            }
        }
```